grafana/grafana
# Prometheus: Respect dashboard queries when querying ad hoc filter labels

**What is this feature?**

When querying ad-hoc filter, we can now access the dashboard queries. So we can use those queries to filter out unnecessary keys that we cannot apply to current metrics. But this works only with the new architecture. For that `dashboardScene` feature flag must be enabled. 

**Why do we need this feature?**

Better ad-hoc filter queries

**Who is this feature for?**

Prometheus users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/84276

**Special notes for your reviewer:**
### How to test
- Simply create a Prometheus dashboard and add an ad-hoc filter template variable
- Create a couple of panels that have different metrics
- Click to add ad-hoc filter and see all the labels are fetched
- Switch to this branch and try again. 
- See only applicable labels are fetched.

Please check that:
